mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/InMemoryMessageIdManager.java

```diff
@@ -140,18 +140,12 @@ public void setInMailboxes(MessageId messageId, List<MailboxId> mailboxIds, Mail
     }
 
     private List<MailboxId> getUsersMailboxIds(MailboxSession mailboxSession) throws MailboxException {
-        return mailboxManager.search(userMailboxes(mailboxSession), mailboxSession)
+        return mailboxManager.search(MailboxQuery.builder().matchesAllMailboxNames().build(), mailboxSession)
```

Comment:
What about a _MailboxQuery.ALL_MAILBOXES_QUERY_ constant?

Reply:
I don't think it make sense, it's clear with builder IMO